Assets/SEE/Game/City/SEECity.cs

```diff
@@ -308,8 +313,14 @@ public virtual async UniTask LoadDataAsync()
                 {
                     using (LoadingSpinner.ShowIndeterminate($"Loading city \"{gameObject.name}\""))
                     {
-                        LoadedGraph = await DataProvider.ProvideAsync(new Graph(""), this, x => ProgressBar = x,
-                                                                      cancellationTokenSource.Token);
+                    	ShowNotification.Info("SEECity", "Loading graph");
+                    	Debug.Log("Loading graph from provider");
+                    	IsPipelineRunning = true;
+                        LoadedGraph = await UniTask.RunOnThreadPool(() => DataProvider.ProvideAsync(new Graph(""), this, x => ProgressBar = x,
+                                                                 cancellationTokenSource.Token));
```

Comment:
I'm not sure if this is really the best idea, to be honest—running on the thread pool has certain limitations (e.g., switching threads is expensive, and access to certain Unity methods/fields is prohibited and will lead to errors). For example, using `ShowNotification` can result in errors when on the thread pool. I think it would be better to let each graph provider handle by itself when it needs to switch to the thread pool (this was why they were made async in the first place). This way, even if the graph provider needs to be on the main thread, it can use e.g. strategic `Yield()`s to not freeze the game.

Perhaps we should coordinate a bit on this issue, since I'm currently working on something similar (making graph drawing async).